src/core/algorithms/gfd/parser.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "gfd.h"
+#include "graph_descriptor.h"
+
+namespace Gfdtools {
+
+class Parser {
+private:
+    static std::vector<std::string> Split(std::string str, std::string sep);
+    static std::vector<Literal> ParseLiterals(std::istream& stream);
+    static void WriteLiterals(std::ostream& stream, const std::vector<Literal>&);
+
+public:
+    static void ReadGraphDot(std::istream& stream, graph_t& result);
+    static void ReadGraphDot(std::string file_name, graph_t& result);
+
+    static void ReadGraph(std::string edges_name, std::string features_name, graph_t& result);
+
+    static void ReadGraph(std::istream& stream, graph_t& result);
+    static void ReadGraph(std::string file_name, graph_t& result);
+
+    static void ReadGraphTxt(std::istream& stream, graph_t& result);
+    static void ReadGraphTxt(std::string file_name, graph_t& result);
+
+    static void WriteGraph(std::ostream& stream, graph_t& result);
+    static void WriteGraph(std::string file_name, graph_t& result);
+
+    static void ReadGfd(std::istream& stream, Gfd& result);
+    static void ReadGfd(std::string file_name, Gfd& result);
+
+    static void WriteGfd(std::ostream& stream, Gfd& result);
+    static void WriteGfd(std::string file_name, Gfd& result);

```

Comment:
1) Prefer using `std::filesystem::path const&` for `file_name`
2) Rename `file_name` to `path` (for consistency with `CSVParser`, for example)
3) Also copy-paste of `ReadGraph...(std::string filename, graph_t& result);` methods looks pretty bad. Maybe it's worth making one function for them?
4) Can we just return the graph instead of accepting it by reference? (https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs)
5) We use this class just to group `static` methods. According to google C++ style guide we should avoid this. (https://google.github.io/styleguide/cppguide.html#Nonmember,_Static_Member,_and_Global_Functions)
6) Why for `static void ReadGraph(std::string edges_name, std::string features_name, graph_t& result);` we use `std::string`? In other functions we use `std::istream`.